### PR TITLE
Fixes #23483: edit Foreman license info so that GitHub recognizes it

### PR DESCRIPTION
**Pull request by eirinikos**
Hello! 👋

(CCing @dankohn who has requested this work so that the license will appear correctly in https://landscape.cncf.io/selected=foreman)

GitHub uses a library called Licensee to identify a project's license
type. It shows this information in the status bar and via the API if it
can unambiguously identify the license.

This commit modifies LICENSE so that it begins with the full text of
the GPL 3.0 license. The text that was previously at the beginning of
LICENSE has been moved to the relevant section of the README, which has
been renamed to "Licensing". At the bottom of the "Licensing" section,
there is a note that mentions to the reader that they can reference
LICENSE to see the full text of not only the GPL 3.0 license, but also
all licenses related to Foreman's dependencies.

Collectively, these changes allow Licensee to successfully identify the
license type of Foreman as GPL 3.0.

Here is the output I get when I run Licensee locally on Foreman's remote repo:
```
$ licensee detect https://github.com/theforeman/foreman
License:        Other
Matched files:  LICENSE, README.md, package.json
LICENSE:
  Content hash:  a17b7fcbdc64255d04d0b05e6642373a8ccdf97f
  License:       Other
  Closest licenses:
    GPL-3.0 similarity:   77.68%
    AGPL-3.0 similarity:  74.68%
    LGPL-2.1 similarity:  53.60%
README.md:
  Content hash:  18f341307338ee217d08193931fa550b30ed563b
  License:       Other
  Closest licenses:
    WTFPL similarity:    5.00%
    MS-RL similarity:    2.34%
    BSL-1.0 similarity:  1.75%
package.json:
  Confidence:  90.00%
  Matcher:     Licensee::Matchers::NpmBower
  License:     GNU General Public License v3.0
```

And here is the output I get when I run Licensee on my local clone of Foreman (on the `update-license` branch):
```
License:        GNU General Public License v3.0
Matched files:  LICENSE, package.json
LICENSE:
  Content hash:  f9aaff304edbbe732f27b554dcb6abe49e0b4fa0
  Attribution:   Copyright (C) 2007 Free Software Foundation, Inc. <http://fsf.org/>
  Confidence:    100.00%
  Matcher:       Licensee::Matchers::Exact
  License:       GNU General Public License v3.0
package.json:
  Confidence:  90.00%
  Matcher:     Licensee::Matchers::NpmBower
  License:     GNU General Public License v3.0
```

### CODE REVIEW

**Review by lzap — DISMISSED** [review]
Thank you, this looks good to me. This is important so will let others to approve before we merge.

File: README.md
Comment (ohadlevy) [inline README.md]: we can probably bump the year to 2018 at this stage too

**Review by ohadlevy — COMMENTED** [review]
this looks good to me, any idea why package.json is only 90% (vs 100?).
I also left an inline comment if we can bump the copyright year - thanks.

**Comment by eirinikos** [comment]
Thanks @lzap and @ohadlevy for the quick response. I pushed a new commit to bump the copyright year to 2018.

GitHub says that I `dismissed lzap’s stale review` when I pushed my new commit - is this OK? The wording seems a bit harsh, ha.

> this looks good to me, any idea why package.json is only 90% (vs 100?).

@ohadlevy I'm not sure why Licensee outputs `Confidence:  90.00%` instead of 100%. But I think it's related to the reliance on a [regex](https://github.com/benbalter/licensee/blob/master/lib/licensee/matchers/npm_bower.rb) that only checks for something resembling `"license": "GPL-3.0"`.

Apart from that regex, Licensee also [relies on the Sørensen–Dice coefficient](https://github.com/benbalter/licensee/tree/master/docs#the-solution) to calculate the similarity between Foreman's `package.json` contents and the contents that it expects (whatever that is).

I tried testing Licensee on a `package.json` that only contained the following:

```
{
  "license": "GPL-3.0",
}
```

I thought that it might output 100% , but it still output only 90%. However, 90% seems to be perfectly fine. 👍

**Review by ohadlevy — APPROVED** [review]

**Pull request merged**
